File: api/inventario-api/DTOs/MovementOutput.cs
using inventario_api.Domain.Shared;

namespace inventario_api.DTOs
{
    public class MovementOutput
    {
        public Guid Id { get; set; }
        public decimal Quantity { get; set; }
        public Guid ProductId { get; set; }
        public string Product { get; set; }
        public MovementType Type { get; set; }
        public DateTime Created { get; set; }
    }
}

File: app/lib/models/movement_output.dart
class MovementOutput {
  final String id;
  final double quantity;
  final String productId;
  final String product;
  final int type;
  final DateTime created;

  MovementOutput({
    required this.id,
    required this.quantity,
    required this.productId,
    required this.product,
    required this.type,
    required this.created,
  });

  factory MovementOutput.fromJson(Map<String, dynamic> json) {
    return MovementOutput(
      id: json['id'],
      quantity: (json['quantity'] as num).toDouble(),
      productId: json['productId'],
      product: json['product'],
      type: json['type'],
      created: DateTime.parse(json['created']),
    );
  }
}